api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
@flx42 Thanks for the pointers! This isn't my area, so thanks for sticking with this.

My guess is that we don't need major/minor granularity. That will likely cause massive portability issues.

> Also, I feel like I'm missing something since the agent is using the engine-api

It is using the engine-api in the swarmkit version but let's not limit it ourselves to that. We need to focus on the requirements for the service model and work back from there. If we have to add something to engine-api, then so be it.

The only other problem I can think of here is requesting a device class versus a specific device. For example, do we need to be able to say "mount anything that starts with `/dev/snd`" or it is sufficient to assume that there is only a single device and we can discovery others from there.